packages/base-map/src/index.tsx

```diff
@@ -0,0 +1,201 @@
+  maxZoom?: number;
+  /** A callback method which is fired when the map is clicked with the left mouse button/tapped */
+  onClick?: (evt: Event) => void;
+  /** A callback method which is fired when the map is clicked with the right mouse button/long tapped */
+  // Unknown is used here because of a maplibre/mapbox issue with the true type, MapLayerMouseEvent
+  onContextMenu?: (e: unknown) => void;
+  /** A callback method which is fired when the map zoom or map bounds change */
+  // TODO: does this cause integration issues?
+  onViewportChanged?: (e: maplibregl.MapLibreEvent) => void;
+  /** A ref to pass to the MapLibre react component */
+  passedRef?: React.Ref<MapRef>;
+  /** An initial zoom value for the map */
+  zoom?: number;
+};
+type State = {
+  fitBoundsOptions?: Record<string, number | string | boolean>;
+  latitude: number;
+  longitude: number;
+  zoom: number;
+};
+
+const BaseMap = ({
+  // These tiles are free to use, but not in production
+  baseLayer = "https://basemaps.cartocdn.com/gl/positron-gl-style/style.json",
+  center,
+  children,
+  forceMaxHeight,
+  mapLibreProps,
+  maxZoom,
+  onClick,
+  onContextMenu,
+  passedRef,
+  onViewportChanged,
+  zoom: initZoom = 12
+}: Props): JSX.Element => {
+  const [viewState, setViewState] = React.useState<State>({
+    fitBoundsOptions: {
+      animate: true,
+      duration: 300,
+      essential: false,
+      maxDuration: 600,
+      padding: 200
+    },
+    latitude: center?.[0],
+    longitude: center?.[1],
+    zoom: initZoom
+  });
+
+  // On mobile hover is unavailable, so we use this variable to use a two tap process
+  // to simulate a hover
+  const [fakeMobileHover, setFakeMobileHover] = useState(false);
+
+  useEffect(() => {
+    callIfValid(onViewportChanged)(viewState);
+  }, [viewState]);
+
+  useEffect(() => {
+    if (center?.[0] === null || center?.[1] === null) return;
+
+    setViewState({
+      ...viewState,
+      latitude: center?.[0],
+      longitude: center?.[1]
+    });
+  }, [center]);
+
+  const toggleableLayers = Array.isArray(children)
+    ? children
+        .flat()
+        .filter(child => child?.props?.id !== undefined)
+        .map(child => {
+          const { id, name, visible } = child.props;
+          return { id, name, visible };
+        })
+    : [];
+
+  const [hiddenLayers, setHiddenLayers] = useState(
+    toggleableLayers.filter(layer => !layer?.visible).map(layer => layer.id)
+  );
+
+  const adjustHiddenLayers = useCallback(
+    id => {
+      const updatedLayers = [...hiddenLayers];
+      // Delete the layer id if present, add it otherwise
+      updatedLayers.includes(id)
+        ? updatedLayers.splice(updatedLayers.indexOf(id), 1)
+        : updatedLayers.push(id);
+
+      setHiddenLayers(updatedLayers);
+    },
+    [hiddenLayers]
+  );
+
+  return (
+    <Map
+      // eslint-disable-next-line react/jsx-props-no-spreading
+      {...mapLibreProps}
+      latitude={viewState.latitude}
```

Comment:
Add and pass in an `id` prop, so that implementers can use the `useMap` hook instead of passing refs around.

Reply:
Even with an id, `useMap` only works when the component is a child of the parent. The id is only useful when managing multiple maps.

Reply:
See these docs for `useMap` outside of a `<Map>` element: https://visgl.github.io/react-map-gl/docs/api-reference/use-map

Reply:
In that example the `NavigateButton` is a child of a `MapProvider` that the Map is also a child of. This architecture doesn't work in our case (I experimented with this which is why that `MapProvider` was originally left behind)